### PR TITLE
Add hascuSOLVER flag to Context

### PR DESCRIPTION
**Pull request by lezcano**
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #70253
* #69827
* #69826
* __->__ #69825

As per title.

cc @ngimel @jianyuh @nikitaved @pearu @mruberry @walterddr @IvanYashchuk @xwang233 @Lezcano

Differential Revision: [D33751986](https://our.internmc.facebook.com/intern/diff/D33751986)

### CODE REVIEW

**Comment by IvanYashchuk** [comment]
Maybe we should also expose `CUSOLVER_VERSION` and `MAGMA_VERSION_MAJOR, MAGMA_VERSION_MINOR, MAGMA_VERSION_MICRO`...

**Comment by Lezcano** [comment]
I'll wait for the CI to pass. If it does, I'll add those things to this PR.

File: aten/src/ATen/Context.h [hunk 1]
```diff
@@ -74,6 +74,9 @@ class TORCH_API Context {
   static long versionCuDNN() {
     return detail::getCUDAHooks().versionCuDNN();
   }
+  static bool hascuSOLVER() {
```
Comment (mruberry) [inline aten/src/ATen/Context.h:77]: I get this should be `hasCuSOLVER` to be consistent with `versionCuDNN` above?

**Review by mruberry — APPROVED** [review]
Seems OK to be but @ngimel is our CUDAHooks expert

**Comment by mruberry** [comment]
@mruberry has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D33202687).

**Review by ngimel — APPROVED** [review]

**Comment by mruberry** [comment]
@mruberry has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D33202687).

**Comment by mruberry** [comment]
@mruberry has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D33202687).

**Comment by mruberry** [comment]
@mruberry has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D33202687).

**Comment by mruberry** [comment]
@mruberry has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D33202687).

**Comment by mruberry** [comment]
@mruberry has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D33202687).

**Comment by mruberry** [comment]
@mruberry has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D33751986).

**Pull request closed without merging**